src/api/worker/rest/RestClient.js

```diff
@@ -141,6 +146,25 @@ export class RestClient {
 		}
 	}
 
+	_saveServerTimeOffsetFromRequest(xhr: XMLHttpRequest) {
+		const serverTimestamp = xhr.getResponseHeader("Date")
+		const serverTime = new Date(serverTimestamp).getTime()
```

Comment:
Are you *sure* it's the correct way to parse that date? HTTP Date is RFC 7231:
https://datatracker.ietf.org/doc/html/rfc7231#section-7.1.1.2

and js Date accepts RFC 2822:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date

and I'm not 100% they are the same

Reply:
It works at least in firefox and chromium. If we can test it in edge and safari as well should we consider it safe? Or we can use luxon, or parse ourselves

Reply:
I think if we can look at RFC and say "it's compatible" it's already enough?

Reply:
We think it makes sense to trust that the format sent by the server will parseable by javascript, as it seems to be the case in general. We could make server tests to ensure that the format being sent is always what is expected

Reply:
I think unless we can prove that they are logically compatible (minus TZ stuff maybe) there's still a chance that it will poop itself

Reply:
rfc7231 provides three options of formats, the preferred one being `IMF-fixdate`. This one is definitely parseable by any rfc2822 compatible parser, since it is a strict subset (with no folding white space) of the format of rfc5322, which is the same as rfc2822 accepting more folding white spaces.

Furthermore, there is no reason to expect the server to return any of the other two accepted formats, which are obsolete and accepted only for backwards compatibility.

I think we're safe here.

Reply:
:clap: :clap: :clap: 

Now I can sleep without fear of Invalid Date Format